test/test_preprocessor.py

```diff
@@ -91,3 +91,18 @@ def test_clean_header_footer():
 
     assert "This is a header." not in documents[0]["content"]
     assert "footer" not in documents[0]["content"]
+
+
+def test_remove_substrings():
+    converter = PDFToTextConverter()
+    document = converter.convert(file_path=Path(SAMPLES_PATH / "pdf" / "sample_pdf_2.pdf"))
```

Comment:
Just a minor thing: the test might be slightly faster if we were simply creating a document on the fly (like `document = Document(content="...content...")`). Is there a special reason to use the PDF?

Reply:
PDF is the format that has causes troubles with odd characters sometimes. But I guess, any utf-8 text format should be the same here. So, good point, thanks!